File: trees/inorderTraversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        arr=[]

        self.fn(root, arr)

        return arr        

    def fn(self, root: Optional[TreeNode], arr):
        if root==None:
            return

        self.fn(root.left, arr)
        arr.append(root.val)
        self.fn(root.right, arr)

File: trees/inorderTraversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> arr;

        fn(root, arr);

        return arr;
    }

    void fn(TreeNode* root, vector<int>& arr){
        if(root==NULL) return;

        fn(root->left, arr);
        arr.push_back(root->val);
        fn(root->right, arr);
    }
};